save-cosv/src/main/kotlin/com/saveourtool/save/cosv/controllers/CosvController.kt

```diff
@@ -169,14 +172,34 @@ class CosvController(
             rawCosvFileStorage.listByOrganization(organizationName)
         }
 
+    /**
+     * @param identifier
+     * @return list of cosv files
+     */
+    @GetMapping("/list-versions")
+    fun listVersions(
+        @RequestParam identifier: String,
+    ): Flux<CosvFileDto> = blockingToFlux {
+        cosvFileRepository.findAllByIdentifier(identifier).map { it.toDto() }
```

Comment:
please do not use `cosvFileRepository` directly, it's internal logic for `cosvStorage`.
Please add this method to `cosvRepository` and add user `cosvStorage` to get list.

Note: `cosvRepository` hides implementation details about storing COSV entries and user `cosvStorage` in current implementation.
`cosvStorage` uses `cosvFileRepository` as a storage for metadata for content saved in S3

Reply:
Fixed